### PR TITLE
Pin nodejs version in Dockerfile

### PR DESCRIPTION
**Pull request by lyw07**
### Summary

This PR is to fix the issue that nodejs version cannot be pinned when installing from PPA, because the pinned version got removed from PPA when a new version is released.

This PR introduces a new environment variable in Dockerfile `NODE_VERSION` to pin the version of nodejs, downloads the deb installer of specific nodejs from https://deb.nodesource.com/node_10.x/pool/main/n/nodejs and installs it.

### Reviewer guidance

Please check if the build runs successfully.

### References

Example error on buildkite: https://buildkite.com/learningequality/kolibri/builds/8795#0f5c102b-e95f-40c9-9f90-1c4d26fd1490

----

### Contributor Checklist

PR process:

- [X] PR has the correct target branch and milestone
- [X] PR has 'needs review' or 'work-in-progress' label
- [X] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Gherkin stories have been updated
- [ ] Unit tests have been updated

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by benjaoming** [comment]
Hi @lyw07 

The solution fits well, I think that what is described about keeping the files themselves in the archive but removing them from the `Releases` repository manifest is indeed what's happening.. and this would be the way to fix it.

> This PR is to fix the issue that nodejs version cannot be pinned when installing from PPA, because the pinned version got removed from PPA when a new version is released.

Nit-pick: This isn't technically speaking a PPA. While the differences are very small, a PPA is a repository that is hosted via Launchpad and has the specific feature of `apt-add-repository` being able to fetch and add the PGP signing keys for that particular repository. Maybe there are other differences, too.. for instance, with a Launchpad PPA, you can only add packages for currently active Ubuntu dists. The node-js repository is actually just a repository, so users have to install the repository keys manually before using it :)

Oh, some more background: Because of a bit of complexities using `apt-add-repository`, I have actually defaulted to using the generic way of manually adding keys. Effectively treating our PPA as a normal repository. This was more of a documentation issue, because I couldn't figure out how to write "but if you are on system B, C, D, please use this method".

File: docker/base.dockerfile [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
-      curl \
-      software-properties-common
-
-# Install nodejs and add 'hold' such that it doesn't get upgraded
-RUN curl -sL https://deb.nodesource.com/setup_10.x | bash -
-
-# add yarn ppa
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
-RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+ENV NODE_VERSION=10.14.1
```
Comment (benjaoming) [inline docker/base.dockerfile:3]: I think this should be in the `env.list` such that we have everything collected neatly in one place :)

File: docker/base.dockerfile [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
 
+# install latest python
```
Comment (benjaoming) [inline docker/base.dockerfile]: Not sure if that comment is very accurate. It wasn't before, neither and that's my fault :)

**Review by benjaoming — APPROVED** [review]
A couple of comments, but feel free to merge. It's a nice and clean solution.

File: docker/base.dockerfile (hunk 1 above)
Comment (lyw07) [reply docker/base.dockerfile:3]: @benjaoming yes I was putting it into env.list first, but it seems like during build time, it doesn't read from env.list file. Also I notice that `docker build` doesn't have the `--env-file` option (https://docs.docker.com/engine/reference/commandline/build/#options). So I guess `env.list` is only available during `docker run` not `docker build`? In this case, we need to have the node version in image build time to download the deb installer, so I feel like putting ENV in the Dockerfile seems to be the only way to make it accessible during build time.

**Comment by lyw07** [comment]
@benjaoming thank you so much for the full explanation! I agree with you that the word "PPA" is not accurate here. It's great to learn something more about "PPA" vs repository!

Also I will push another commit to fix the comment issue you mentioned above. Thanks!

**Pull request merged**